Gems/ROS2/Code/Source/Utilities/ROS2Names.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2/Utilities/ROS2Names.h"
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/std/string/regex.h>
+#include <rcl/validate_topic_name.h>
+#include <rmw/validate_namespace.h>
+
+namespace ROS2
+{
+    AZStd::string ROS2Names::GetNamespacedName(const AZStd::string& ns, const AZStd::string& name)
+    {
+        if (ns.empty())
+        {
+            return name;
+        }
+
+        return AZStd::string::format("%s/%s", ns.c_str(), name.c_str());
+        ;
+    }
+
+    AZStd::string ROS2Names::RosifyName(const AZStd::string& input)
+    {
+        // TODO - add unit tests
+        // TODO - implement stricter guidelines and differentiate: https://design.ros2.org/articles/topic_and_service_names.html
```

Comment:
It looks like 2 github issues should be created for adding unit test and to implement stricter guidelines?